### PR TITLE
Docstrings for `VideoClassificationData`

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

Part of #957 

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/video/classification/data.py [hunk 1]
```diff
@@ -47,9 +47,14 @@
 else:
     ClipSampler = None
 
+# Skip doctests if requirements aren't available
+if not _VIDEO_AVAILABLE:
+    __doctest_skip__ = ["VideoClassificationData", "VideoClassificationData.*"]
```
Comment (tchaton) [inline flash/video/classification/data.py:52]: Neat !

File: flash/video/classification/data.py [hunk 2]
```diff
@@ -186,10 +490,10 @@ def from_data_frame(
             decoder=decoder,
         )
 
-        train_data = (train_data_frame, input_field, target_fields, train_images_root, train_resolver)
-        val_data = (val_data_frame, input_field, target_fields, val_images_root, val_resolver)
-        test_data = (test_data_frame, input_field, target_fields, test_images_root, test_resolver)
-        predict_data = (predict_data_frame, input_field, predict_images_root, predict_resolver)
+        train_data = (train_data_frame, input_field, target_fields, train_videos_root, train_resolver)
```
Comment (tchaton) [inline flash/video/classification/data.py:493]: Good catch !

**Review by tchaton — APPROVED** [review]
Awesome work !

File: flash/video/classification/data.py [hunk 3]
```diff
@@ -121,6 +204,112 @@ def from_folders(
...
+            >>> import os
+            >>> import torch
+            >>> from torchvision import io
+            >>> data = torch.randint(255, (10, 64, 64, 3))
+            >>> os.makedirs(os.path.join("train_folder", "cat"), exist_ok=True)
+            >>> os.makedirs(os.path.join("train_folder", "dog"), exist_ok=True)
+            >>> os.makedirs("predict_folder", exist_ok=True)
+            >>> io.write_video(os.path.join("train_folder", "cat", "video_1.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> io.write_video(os.path.join("train_folder", "dog", "video_2.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> io.write_video(os.path.join("train_folder", "cat", "video_3.mp4"), data, 5, "libx264rgb", {"crf": "0"})
```
Comment (Borda) [inline flash/video/classification/data.py:279]:
```suggestion
            >>> _ = [io.write_video(os.path.join("train_folder", lb, f"video_{i + 1}.mp4"), data, 5, "libx264rgb", {"crf": "0"}) for i, lb in enumerate(["cat", "dog", "cat"])]
```

File: flash/video/classification/data.py [hunk 4]
```diff
@@ -174,6 +363,121 @@ def from_data_frame(
...
+
+            >>> import os
+            >>> import torch
+            >>> from torchvision import io
+            >>> data = torch.randint(255, (10, 64, 64, 3))
+            >>> os.makedirs("train_folder", exist_ok=True)
+            >>> os.makedirs("predict_folder", exist_ok=True)
+            >>> io.write_video(os.path.join("train_folder", "video_1.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> io.write_video(os.path.join("train_folder", "video_2.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> io.write_video(os.path.join("train_folder", "video_3.mp4"), data, 5, "libx264rgb", {"crf": "0"})
```
Comment (Borda) [inline flash/video/classification/data.py:430]:
```suggestion
            >>> _ = [io.write_video(os.path.join("train_folder", f"video_{i}.mp4"), data, 5, "libx264rgb", {"crf": "0"}) for i in range(1, 4)]
```

File: flash/video/classification/data.py [hunk 5]
```diff
@@ -174,6 +363,121 @@ def from_data_frame(
...
+            ...     transform_kwargs=dict(image_size=(244, 244)),
+            ...     batch_size=2,
+            ... )
+            >>> datamodule.num_classes
+            2
+            >>> datamodule.labels
+            ['cat', 'dog']
+            >>> model = VideoClassifier(backbone="x3d_xs", num_classes=datamodule.num_classes)
+            >>> trainer = Trainer(fast_dev_run=True)
+            >>> trainer.fit(model, datamodule=datamodule)  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
```
Comment (Borda) [inline flash/video/classification/data.py:469]:
```suggestion
            >>> trainer.fit(model, datamodule=datamodule)  # doctest: +ELLIPSIS
```
with Doctest-plus this `NORMALIZE_WHITESPACE` shall be default

File: flash/video/classification/data.py [hunk 6]
```diff
@@ -231,6 +535,137 @@ def from_csv(
...
+            >>> import os
+            >>> import torch
+            >>> from torchvision import io
+            >>> from pandas import DataFrame
+            >>> data = torch.randint(255, (10, 64, 64, 3))
+            >>> os.makedirs("train_folder", exist_ok=True)
+            >>> os.makedirs("predict_folder", exist_ok=True)
+            >>> io.write_video(os.path.join("train_folder", "video_1.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> io.write_video(os.path.join("train_folder", "video_2.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> io.write_video(os.path.join("train_folder", "video_3.mp4"), data, 5, "libx264rgb", {"crf": "0"})
```
Comment (Borda) [inline flash/video/classification/data.py:603]:
```suggestion
            >>> _ = [io.write_video(os.path.join("train_folder", f"video_{i}.mp4"), data, 5, "libx264rgb", {"crf": "0"}) for i in range(1, 4)]
```
just wondering if there can be `testsetup` common for all modules as for example, you create the same files for each test and in fact you are fine to create them just for module start and clean at the end

**Review by Borda — APPROVED** [review]

File: flash/video/classification/data.py (hunk 5 above)
Comment (ethanwharris) [reply flash/video/classification/data.py:469]: Yes but not within PyCharm doctest runner, so at present the tests will pass from PyCharm

File: flash/video/classification/data.py (hunk 6 above)
Comment (ethanwharris) [reply flash/video/classification/data.py:603]: Eventually these will be moved to `tests/helpers` so you get a good balance of readability and simplicity, but this will be done in a follow-up when we overhaul the tests :smiley:
